### PR TITLE
Testing the messaging and fix of previous tests

### LINKED ISSUE #188: Fix the tests that fail because of the chatviewmodel (ConfirmedLessonTest & OverviewTest)

### PR DESCRIPTION
**Pull request by Philipp0206**
# 🎯 PR Description:
This pull request fixes of 2 previous tests that failed due to the chatviewmodel. I also modified the 2 current endToEndTests. After some long struggles over mutiple weeks with the tests of the messaging feature especially with the mocking of the chatClient. I tried with multiple teamates to test this feature but came to the conclsuion that I'm not sure if the chat can be tested nor should as some teamates mentioned. 

# 📁 Files Changed:

📂 StudentToEndTest.kt
Removed bad comments and updated the name of the variable of the currentChannelId.

📂 TutorEndToEnd.kt
Modified the file and therefore had to add the imports. Added the mock of the chatviewmodel.

📂 ConfirmedLessonTest.kt
Removed the mock of the chatclient and added a good version of the chatviewmodel. Removed the donothing() as it is now fixed in the chatviewmodel. Removed the SMS test as it does not work anymore due to the new messaging way.

📂 OverviewTest.kt
Modified the file by adding the chatviewmodel.

📂 ChatViewModel.kt
Updated the connectUser function so that the failing tests pass as it was an edge case that can't happen while running. Only the mock can have chatClient that is null.

📂 Channel.kt
Upgraded the file for better quality by using the navigationItems variable. 

📂 gradle/libs.versions.toml
Included some mockito necessary for the chatviewmodel.

📂 gradle.kts.kt
Just rearranged the file as I modified it multiple time. But nothing added in the end.

Close #188

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt:1]: Thanks for the setup. More substantial changes to this coming in my PR

File: app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt:1]: Same as above, bigger changes coming in Alessandro's PR

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:1]: Thanks for fixing these! They also seem to pass the CI reliably so I guess our changes worked!

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt [hunk 1]
```diff
@@ -165,19 +196,11 @@ class ConfirmedLessonTest {
...
     composeTestRule.onNodeWithTag("backButton").performClick()
     verify(mockNavigationActions).goBack()
-  }*/
-
-  @Test
-  fun confirmedLessonScreenOpensSmsApp() {
-    composeTestRule.waitForIdle()
-
-    // Click on the "Message Tutor/Student" button
-    composeTestRule.onNodeWithTag("contactButton").performClick()
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:180]: Good initiative to remove unused commented code

File: app/src/androidTest/java/com/github/se/project/ui/overview/OverViewTest.kt [hunk 2]
```diff
@@ -1,363 +1,398 @@
...
+  @Before
+  fun setup() {
+    /// Mock the dependencies
+    profilesRepository = mock(ProfilesRepository::class.java)
+    lessonRepository = mock(LessonRepository::class.java)
+    navController = mock()
+
+    // Initialize the ViewModel with a spy
+    listProfilesViewModel = ListProfilesViewModel(profilesRepository)
+    listProfilesViewModel = spy(listProfilesViewModel)
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/overview/OverViewTest.kt:182]: Very good idea to use a spy! Definitely a feature of Mockito we have underutilised

File: app/src/androidTest/java/com/github/se/project/ui/overview/OverViewTest.kt
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/overview/OverViewTest.kt:1]: Nice fixes here too! I don't know why Git decided to make the diff this confusing, but this should be a great improvement to coverage!

File: app/src/main/java/com/github/se/project/model/chat/ChatViewModel.kt [hunk 3]
```diff
@@ -17,7 +17,7 @@ import kotlinx.coroutines.flow.asStateFlow
  * This ViewModel interacts with the Stream Chat SDK to manage the connection of users, and channel
  * creation/retrieval for specific lessons between tutors and students.
  */
-class ChatViewModel(private val chatClient: ChatClient) : ViewModel() {
+open class ChatViewModel(private val chatClient: ChatClient) : ViewModel() {
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/model/chat/ChatViewModel.kt:20]: Good change, makes mocking easier

File: app/src/main/java/com/github/se/project/ui/message/Channel.kt [hunk 4]
```diff
@@ -70,12 +68,7 @@ fun ChannelScreen(
             bottomBar = {
               BottomNavigationMenu(
                   onTabSelect = { route -> navigationActions.navigateTo(route) },
-                  tabList =
-                      when (currentProfile?.role) {
-                        Role.TUTOR -> LIST_TOP_LEVEL_DESTINATIONS_TUTOR
-                        Role.STUDENT -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
-                        else -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
-                      },
+                  tabList = navigationItems,
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/ui/message/Channel.kt:71]: Much cleaner implementation!

File: gradle/libs.versions.toml
Comment (A-M-00) [inline gradle/libs.versions.toml:1]: Ah, well done! This fixes an issue some of us had with gradle builds

File: app/src/androidTest/java/com/github/se/project/ui/overview/OverViewTest.kt [hunk 5]
```diff
@@ -1,363 +1,398 @@
...
-        composeTestRule.setContent {
-            HomeScreen(
-                listProfilesViewModel,
-                lessonViewModel,
-                networkStatusViewModel,
-                chatViewModel,
-                navigationActions)
-        }
+    // Stub the methods in ChatViewModel
+    // doNothing().`when`(chatViewModel).connectUser(any())
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/overview/OverViewTest.kt:203]: Why are these commented out? If they're not needed, they would be better off being removed

**Review by A-M-00 — APPROVED** [review]
Pretty useful PR overall, fixing these tests should bump up the coverage a fair bit
